hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/map/MapIndexScanPMigrationStressTest.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2021 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.connector.map;
+
+import com.hazelcast.client.test.TestHazelcastFactory;
+import com.hazelcast.config.IndexConfig;
+import com.hazelcast.config.IndexType;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.jet.core.JetTestSupport;
+import com.hazelcast.jet.sql.SqlTestSupport.Row;
+import com.hazelcast.map.IMap;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.SlowTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({SlowTest.class, ParallelJVMTest.class})
+public class MapIndexScanPMigrationStressTest extends JetTestSupport {
```

Comment:
If I slightly modify the tests (repeatedly stopping random member, executing multiple queries) they fail with various cluster related exceptions. Is it expected?

Reply:
Yes, it is expected. 
The main purpose of this test is to launch a query which reads big amount during cluster state change.
If you would execute multiple queries, it can fail with cluster-related exceptions. It happens because at the moment cluster don't wait the jobs finish before shutdown. It is expected to fix it rolling upgrades part 2 PR.